lib/ParseSIL/ParseSIL.cpp

```diff
@@ -5372,6 +5379,47 @@ bool SILParserTUState::parseDeclSILStage(Parser &P) {
   return false;
 }
 
+/// Lookup a global variable declaration from its demangled name.
+///
+/// A variable declaration exists for all sil_global variables defined in
+/// Swift. A Swift global defined outside this module will be exposed
+/// via an addressor rather than as a sil_global. Globals imported
+/// from clang will produce a sil_global but will not have any corresponding
+/// VarDecl.
+///
+/// FIXME: lookupGlobalDecl() can handle collisions between private or
+/// fileprivate global variables in the same SIL Module, but the typechecker
+/// will still incorrectly diagnose this as an "invalid redeclaration" and give
+/// all but the first declaration an error type.
+static Optional<VarDecl *> lookupGlobalDecl(Identifier GlobalName,
+                                            SILLinkage GlobalLinkage,
+                                            SILType GlobalType, Parser &P) {
+  std::string GlobalDeclName = Demangle::demangleSymbolAsString(
+      GlobalName.str(),
+      Demangle::DemangleOptions::SimplifiedUIDemangleOptions());
```

Comment:
Still relevant, please make your own set of demangling options.

Reply:
I meant to ask about that. I think these are the right options:
https://github.com/apple/swift/pull/16834/commits/8d9207e2d8468f800ae27c6e832e405378de90de